### PR TITLE
Items must be properties or removing them doesn't cause re-render

### LINKED ISSUE #739: Pressing the remove button on a literal input does not rerender.

### PR DESCRIPTION
**Pull request by jcoyne**
Fixes #739

### CODE REVIEW

File: src/components/editor/property/InputLiteral.jsx [hunk 1]
```diff
@@ -25,23 +25,16 @@ export class InputLiteral extends Component {
...
-  componentDidMount = () => {
-    // Note that it is possible that the resource template hasn't been loaded yet, hence there will not be a property template.
-    if (this.props.propertyTemplate
-        && this.props.propertyTemplate.repeatable === 'false'
-        && this.props.formData !== undefined
-        && this.props.formData.items.length > 0) {
-      this.setState({ disabled: true })
-    }
+  disabled = () => {
+    return this.props.propertyTemplate.repeatable === 'false' &&
```
Comment (jgreben) [inline src/components/editor/property/InputLiteral.jsx]: @jcoyne I think we both worked on refactoring the same thing, but I see here that we do not need the componentDidMount method anymore... How about using this instead (import from Utilities) which uses safe navigation for `repeatable`?

`!booleanPropertyFromTemplate(this.props.propertyTemplate, 'repeatable', true)`

Comment (jcoyne) [reply src/components/editor/property/InputLiteral.jsx]: This was already here, it's not something I changed in this PR. I try not to change more than is absolutely necessary. If you want to change it in another PR, please go ahead.

**Comment by jgreben** [comment]
> Fixes #739

This sould also fix #736

File: src/components/editor/property/InputLiteral.jsx (hunk 1 above)
Comment (jgreben) [reply src/components/editor/property/InputLiteral.jsx]: Ok when this is out of draft I will merge it and then rebase.

File: src/components/editor/property/InputLiteral.jsx [hunk 2]
```diff
@@ -64,15 +55,7 @@ export class InputLiteral extends Component {
...
-  notRepeatableAfterUserInput = (userInputArray, currentcontent) => {
-    let newId = null
-
-    if (this.props.formData === undefined || this.props.formData.items < 1) {
-      newId = this.addUserInput(userInputArray, currentcontent)
-      this.setState({ disabled: true })
-    }
-    return newId
-  }
+  notRepeatableAfterUserInput = (userInputArray, currentcontent) => this.addUserInput(userInputArray, currentcontent)
```
Comment (jgreben) [inline src/components/editor/property/InputLiteral.jsx]: Is this necessary anymore? If it just calls `addUserInput()` then can line 78 just call `addUserInput()` regardless of whether the field is repeatable?

Comment (jcoyne) [reply src/components/editor/property/InputLiteral.jsx]: Thanks. That's a good suggestion. I've made that change.

**Pull request merged**